dropwizard-jersey/src/main/java/io/dropwizard/jersey/DropwizardResourceConfig.java

```diff
@@ -110,6 +112,8 @@ public void setUrlPattern(String urlPattern) {
 
     public String getEndpointsInfo() {
         final StringBuilder msg = new StringBuilder(1024);
+        final Set<EndpointLogLine> endpointLogLines = Sets.newTreeSet(new EndpointComparator());
```

Comment:
Do we really need a `TreeSet`? 
Basically we don't have duplicates and don't care about speed of lookup. The only thing that we need, it's a sorted collection. We could use an ordinary `ArrayList` with some expected size to avoid resizing and then sort it. The collection is relatively small, so sorting should be fast. `TreeSet` has a very big memory footprint and it's not cache-friendly. I would not use it, if we have alternatives.